pmd-groovy/src/main/java/net/sourceforge/pmd/cpd/GroovyTokenizer.java

```diff
@@ -22,7 +24,7 @@
     public void tokenize(SourceCode sourceCode, Tokens tokenEntries) {
         StringBuilder buffer = sourceCode.getCodeBuffer();
 
-        GroovyLexer lexer = new GroovyLexer(IOUtils.toInputStream(buffer.toString()));
+        GroovyLexer lexer = new GroovyLexer(IOUtils.toInputStream(buffer.toString(), StandardCharsets.UTF_8));
```

Comment:
That's one of the two changes, that are outside of test code.
The change here looks correct. However, I don't know, which Charset GroovyLexer is using or expecting...

So, I'd rather use the constructor, that takes a Reader:

    GroovyLexer lexer = new GroovyLexer(new StringReader(buffer.toString()));

Reply:
Done.